lib/puppet/pops/evaluator/evaluator_impl.rb

```diff
@@ -942,7 +942,7 @@ def eval_CallMethodExpression(o, scope)
 
     obj = receiver[0]
     receiver_type = Types::TypeCalculator.infer(obj)
-    if receiver_type.is_a?(Types::PObjectType)
+    if receiver_type.is_a?(Types::PObjectType) || receiver_type.is_a?(Types::PObjectTypeExtension)
```

Comment:
The fact that this is needed shows that there is a fundamental problem. It should be enough to have one test. Easy to make this mistake otherwise. The PObjectTypeExtension is also not just an internal concern since you get it back from inference. I think they need to both `is_a` to a common thing "being a kind of object type". Documentation in PObjectType needs to also point out to not test against `is_a`  unless making a distinction between inferred / not-inferred kinds of object types.

Reply:
And it will be enough with one test when my next PR for URI arrives. It's dealt with there.

Reply:
Thomas pointed out that the next PR (for URI) indeed changes this - just too much work to move the rest of the impl of that over to this PR which just fixes a particular problem. Merging...